### PR TITLE
added a readme to the file

### PR DESCRIPTION
**Pull request by chris-chen-creates**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,46 +1,35 @@
...
+
+# Setup
+1. Download the .zip file or clone the repo using the command line tool:
+```
+gh repo clone chris-chen-creates/todooster
+```
+2. Install the dependencies using:
+```
+npm install
+```
```
Comment (angles-n-daemons) [inline README.md]: Having run instructions would be helpful too! In addition to screenshots and instructions for usage

File: src/App.tsx [hunk 2]
```diff
@@ -1,18 +1,59 @@
...
+import TodoAdd from './components/TodoAdd';
+import { v4 as uuidv4 } from 'uuid';
+
+const App: React.FC = () => {
+  const [todos, setTodo] = useState<TodoProps[]>([]);
+
+  // this is redundant
+  // how can I write this to not mutate the value in place
+  const handleTaskComplete = (id: string) => {
+    const newTodoState: TodoProps[] = [...todos];
```
Comment (angles-n-daemons) [inline src/App.tsx]: Though confusing, I think this syntax functionally does a copy - so you're not technically mutating the value in place

**Review by angles-n-daemons — COMMENTED** [review]
Coupla comments

File: src/App.tsx (hunk 2 above)
Comment (angles-n-daemons) [reply src/App.tsx:16]: Oh you mean not setting the `isComplete` property, but instead creating a new task - also a little confusing, but you could do something like:

```ts
const index = newTodoState.findIndex((todo) => todo.id === id)
const todo = newTodoState[index]
newTodoState[index] = Object.assign({ isComplete: !todo.isComplete}, todo) // creates a copy
```

ooor if you wanna really be strict about immutability you can do:

```ts
setTodo(
    ...newTodoState.slice(0, index),
    Object.assign({ isComplete: !todo.isComplete }, todo),
   ...newTodoState.slice(index + 1),
);
```

Or if you want to use the more clever language syntax you can try the approach [suggested here](https://stackoverflow.com/a/38061256):
```ts
setTodo(
   Object.assign([], newTodoState, {
      index: Object.assign({ isComplete: !todo.isComplete }, todo),
   }),
)
```
For the purest example of immutability 😭

File: src/interface.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
```
Comment (angles-n-daemons) [inline src/interface.ts]: In practice, creating files or directories named `types...` or `interfaces...` doesn't scale well. If it's small and well contained within the package it's used it should be fine, but at `src` it's bound to be an early candidate for cleanup

File: src/App.tsx (hunk 2 above)
Comment (angles-n-daemons) [reply src/App.tsx]: Ah I see, you manually set the `isComplete` property. I think that's probably fine - in practice if you wanted to change TodoProps to a class you could add factory functions to it.

File: src/interface.ts (hunk 3 above)
Comment (chris-chen-creates) [reply src/interface.ts]: So for scaling, would it just be best to add the interfaces/types to whichever file they're being used in?

Comment (angles-n-daemons) [reply src/interface.ts]: Kind of, my philosophy is to define things close to where they're used

Comment (chris-chen-creates) [reply src/interface.ts]: Got that refactored in my last commit

File: README.md [hunk 4]
```diff
@@ -1,46 +1,90 @@
...
+- [Express](https://expressjs.com/)
 
-The page will reload if you make edits.\
-You will also see any lint errors in the console.
+# Setup :hatching_chick:
+1. Download the .zip file or clone the repo using the command line tool:
+```
+gh repo clone chris-chen-creates/todooster
+```
+2. Install the dependencies using:
```
Comment (angles-n-daemons) [inline README.md:34]: Maybe one more comment about migrating the database

File: README.md [hunk 5]
```diff
@@ -1,46 +1,90 @@
...
+```
+npm install
+```
 
-### `yarn test`
+# Running the app
+1. Start by navigating to the `/todooster` folder
+2. In the command line, run either `npm start` or `yarn start` (if you're a yarn user)
+3. *login/registration steps will go here when ready*
+4.
```
Comment (angles-n-daemons) [inline README.md]: Extra step?

File: README.md [hunk 6]
```diff
@@ -1,46 +1,90 @@
...
+**Frontend**
+After a user has marked a task as complete, it'll then be eligible for deletion. When the "Delete Completed" button is pressed, the completed tasks will be removed.
 
-To learn React, check out the [React documentation](https://reactjs.org/).
+**Backend** 
+Deleting tasks will add a timestamp to the deleted field on the associated task in the database. These tasks will no longer be available to the user. 
+
+## Backend Design
+
+### ERD
```
Comment (angles-n-daemons) [inline README.md:91]: Thank you! 🎉

**Review by angles-n-daemons — COMMENTED** [review]
Couple of open comments!

File: README.md (hunk 4 above)
Comment (chris-chen-creates) [reply README.md:34]: donezo!

File: README.md (hunk 5 above)
Comment (chris-chen-creates) [reply README.md]: Was adding as a reminder to finish the login steps. Removed for now.

**Pull request closed without merging**